hooks/open-telemetry/src/main/java/dev/openfeature/contrib/hooks/otel/OpenTelemetryMetricHook.java

```diff
@@ -0,0 +1,167 @@
+import io.opentelemetry.api.metrics.Meter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static dev.openfeature.contrib.hooks.otel.OtelCommons.ERROR_KEY;
+import static dev.openfeature.contrib.hooks.otel.OtelCommons.REASON_KEY;
+import static dev.openfeature.contrib.hooks.otel.OtelCommons.flagKeyAttributeKey;
+import static dev.openfeature.contrib.hooks.otel.OtelCommons.providerNameAttributeKey;
+import static dev.openfeature.contrib.hooks.otel.OtelCommons.variantAttributeKey;
+
+
+/**
+ * OpenTelemetry metric hook records metrics at different {@link Hook} stages.
+ */
+@Slf4j
+@SuppressWarnings("PMD.TooManyStaticImports")
+public class OpenTelemetryMetricHook implements Hook {
+
+    private static final String METER_NAME = "go.openfeature.dev";
+    private static final String EVALUATION_ACTIVE_COUNT = "feature_flag.evaluation_active_count";
+    private static final String EVALUATION_REQUESTS_TOTAL = "feature_flag.evaluation_requests_total";
+    private static final String FLAG_EVALUATION_SUCCESS_TOTAL = "feature_flag.evaluation_success_total";
+    private static final String FLAG_EVALUATION_ERROR_TOTAL = "feature_flag.evaluation_error_total";
+
+    private final LongUpDownCounter activeFlagEvaluationsCounter;
+    private final LongCounter evaluationRequestCounter;
+    private final LongCounter evaluationSuccessCounter;
+    private final LongCounter evaluationErrorCounter;
+    private final List<DimensionDescription> dimensionDescriptions;
+
+    /**
+     * Construct a metric hook by providing an {@link OpenTelemetry} instance.
+     */
+    public OpenTelemetryMetricHook(final OpenTelemetry openTelemetry) {
+        this(openTelemetry, Collections.emptyList());
+    }
+
+    /**
+     * Construct a metric hook with {@link OpenTelemetry} instance and a list of {@link DimensionDescription}.
+     * Provided dimensions are attempted to be extracted from flagMetadata attached to {@link FlagEvaluationDetails}.
+     */
+    public OpenTelemetryMetricHook(final OpenTelemetry openTelemetry, final List<DimensionDescription> dimensions) {
+        final Meter meter = openTelemetry.getMeter(METER_NAME);
+
+        activeFlagEvaluationsCounter =
+                meter.upDownCounterBuilder(EVALUATION_ACTIVE_COUNT).setDescription("active flag evaluations counter")
+                        .build();
+
+        evaluationRequestCounter = meter.counterBuilder(EVALUATION_REQUESTS_TOTAL)
+                .setDescription("feature flag evaluation request counter").build();
+
+        evaluationSuccessCounter = meter.counterBuilder(FLAG_EVALUATION_SUCCESS_TOTAL)
+                .setDescription("feature flag evaluation success counter").build();
+
+        evaluationErrorCounter = meter.counterBuilder(FLAG_EVALUATION_ERROR_TOTAL)
+                .setDescription("feature flag evaluation error counter").build();
```

Comment:
Are there metric standard conventions in OTel that allow us to standardize these descriptions?

I noticed these don't use capitals/punctualtion (both here and in Go). Personally, I like that, but I'm wondering if there's a recommended style?

Reply:
This is optional and does not have any specific format [1]

> The description is an optional free-form text provided by the author of the instrument. The API MUST treat it as an opaque string.

So I don't think we need any formatting here

[1] - https://opentelemetry.io/docs/specs/otel/metrics/api/#instrument-description

Reply:
Cool. I'm fine with keeping them as is, since they are sentence fragments. If anyone has any strong preference for capitalization/punctuation, I will let them make that argument.

Reply:
Thank you. I am open for any suggestions here :)